shell/server/core.js

```diff
@@ -112,33 +116,35 @@ NotificationHandle.prototype.close = function () {
   });
 };
 
-NotificationHandle.prototype.save = function (params) {
-  var self = this;
+saveFrontendRef = function (frontendRef, owner) {
   return inMeteor(function () {
-    var sturdyRef = new Buffer(Random.id(20));
+    var sturdyRef = new Buffer(generateSturdyRef());
     var hashedSturdyRef = hashSturdyRef(sturdyRef);
     ApiTokens.insert({
       _id: hashedSturdyRef,
-      frontendRef: {
-        notificationHandle: self.notificationId
-      },
-      owner: params.sealFor,
+      frontendRef: frontendRef,
+      owner: owner,
+      created: new Date()
     });
-    self.saved = true;
     return {sturdyRef: sturdyRef};
   });
 };
 
-function restoreInternal(sturdyRef, ownerPattern) {
-  // Restores `sturdyRef`, checking first that its owner matches `ownerPattern`.
+NotificationHandle.prototype.save = function (params) {
+  return saveFrontendRef({notificationHandle: this.notificationId}, params.sealFor);
+};
 
+restoreInternal = function (sturdyRef, ownerPattern, requiredPermissions) {
```

Comment:
Can you take a look at e7660d3 and see if that answers your question? requiredPermissions is being passed in to set a constraint, and seems necessary based on how we designed `restore`